soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -470,34 +392,39 @@ namespace Logic {
-           (itemName == SLINGSHOT         && Slingshot)       ||
-           (itemName == BOOMERANG         && Boomerang)       ||
-           (itemName == KOKIRI_SWORD      && KokiriSword)     ||
-           (itemName == STICKS            && Sticks)          ||
-           (itemName == DEKU_SHIELD       && DekuShield)      ||
-           (itemName == FIRE_ARROWS       && FireArrows)      ||
-           (itemName == ICE_ARROWS        && IceArrows)       ||
-           (itemName == LIGHT_ARROWS      && LightArrows);
+    return (itemName == DINS_FIRE              && DinsFire)            ||
+           (itemName == FARORES_WIND           && FaroresWind)         ||
+           (itemName == NAYRUS_LOVE            && NayrusLove)          ||
+           (itemName == LENS_OF_TRUTH          && LensOfTruth)         ||
+           (itemName == BOW                    && Bow)                 ||
+           (itemName == MEGATON_HAMMER         && Hammer)              ||
+           (itemName == IRON_BOOTS             && IronBoots)           ||
+           (itemName == HOVER_BOOTS            && HoverBoots)          ||
+           (itemName == HOOKSHOT               && Hookshot)            ||
+           (itemName == LONGSHOT               && Longshot)            ||
+           (itemName == SILVER_GAUNTLETS       && SilverGauntlets)     ||
+           (itemName == GOLDEN_GAUNTLETS       && GoldenGauntlets)     ||
+           (itemName == GORON_TUNIC            && GoronTunic)          ||
+           (itemName == ZORA_TUNIC             && ZoraTunic)           ||
+           (itemName == SCARECROW              && Scarecrow)           ||
+           (itemName == DISTANT_SCARECROW      && DistantScarecrow)    ||
+           (itemName == HYLIAN_SHIELD          && HylianShield)        ||
+           (itemName == MIRROR_SHIELD          && MirrorShield)        ||
+           (itemName == MASTER_SWORD           && MasterSword)         ||
+           (itemName == BIGGORON_SWORD         && BiggoronSword)       ||
+           (itemName == SLINGSHOT              && Slingshot)           ||
+           (itemName == BOOMERANG              && Boomerang)           ||
+           (itemName == KOKIRI_SWORD           && KokiriSword)         ||
+           (itemName == STICKS                 && Sticks)              ||
+           (itemName == DEKU_SHIELD            && DekuShield)          ||
+           (itemName == FIRE_ARROWS            && FireArrows)          ||
+           (itemName == ICE_ARROWS             && IceArrows)           ||
+           (itemName == LIGHT_ARROWS           && LightArrows)         ||
+           (itemName == OCARINA_A_BUTTON       && OcarinaAButton)      ||
+           (itemName == OCARINA_C_LEFT_BUTTON  && OcarinaCLeftButton)  ||
+           (itemName == OCARINA_C_RIGHT_BUTTON && OcarinaCRightButton) ||
+           (itemName == OCARINA_C_DOWN_BUTTON  && OcarinaCDownButton)  ||
+           (itemName == OCARINA_C_UP_BUTTON    && OcarinaCUpButton);
```

Comment:
You need to have the songs here, not the buttons, because CanUse first uses this to see if you have the item bbeing checked before running the rest of the logic. With this code, CanUse(ZELDAS_LULLABY) will never be true because HasItem isn't checking to see you have ZELDAS_LULLABY.

Reply:
Fixed